### PR TITLE
implement websocket for verification

### PR DESCRIPTION
**Pull request by altayaydemir**
haven't tried to make a payment yet, will do tmrrw 😴 

---

https://cln.sh/KL07CD 👍

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,12 +43,13 @@ To be able to initiate the payment process, you **must create a new payment sess
 ```ts
 // 1- create a new payment session
 type CreateSessionParams = {
-  apiURL: string // the URL of your Accept NANO server
+  apiHost: string // host of your Accept NANO server
```
Comment (altayaydemir) [inline README.md:46]: to derive wss url from the host

File: src/webSocket.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  baseURL: string
+  paymentToken: AcceptNanoPayment['token']
+}) => `${baseURL}?token=${paymentToken}`
+
+type AcceptNanoWebSocketEvents = {
+  open: () => void
+  close: () => void
+  error: (error: unknown) => void
+  payment_updated: (payment: AcceptNanoPayment) => void
+  payment_completed: (payment: AcceptNanoPayment) => void
```
Comment (altayaydemir) [inline src/webSocket.ts:22]: maybe it's better to call this `payment_verified` and get rid of `completed` keyword altogether. we have both `completed payment` and `verified payment` notions but they're identical lol wtf

File: src/types.ts [hunk 3]
```diff
@@ -25,7 +30,16 @@ export interface AcceptNanoPayment {
...
+): input is AcceptNanoPayment => {
+  if (typeof input !== 'object' || !input) {
+    return false
+  }
+
+  const record = input as Record<string, unknown>
+  return Boolean(record.token && record.account && record.currency)
+}
+
+export const isCompletedAcceptNanoPayment = (input: unknown) =>
```
Comment (altayaydemir) [inline src/types.ts:44]: perhaps rename as `isVerifiedAcceptNanoPayment`

File: src/webSocket.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        return isCompletedAcceptNanoPayment(payload)
+          ? eventEmitter.emit('payment_completed', payload)
+          : eventEmitter.emit('payment_updated', payload)
+      }
+
+      logger.log('websocket', 'could not cast payload to payment object', {
+        event,
+        payload,
+      })
+    } catch (error) {
```
Comment (altayaydemir) [inline src/webSocket.ts:46]: add
`eventEmitter.emit('error', error)`

**Review by BatuAksoy — APPROVED** [review]

**Pull request merged**